rclpy/rclpy/__init__.py

```diff
@@ -58,18 +58,19 @@ def create_node(node_name):
     return Node(node_handle)
 
 
-def spin_once(node, timeout_sec=None):
+def spin_once(node, timeout_sec=-1):
     wait_set = _rclpy.rclpy_get_zero_initialized_wait_set()
 
     _rclpy.rclpy_wait_set_init(wait_set, len(node.subscriptions), 0, 0)
 
     _rclpy.rclpy_wait_set_clear_subscriptions(wait_set)
     for subscription in node.subscriptions:
         _rclpy.rclpy_wait_set_add_subscription(wait_set, subscription.subscription_handle)
-    if timeout_sec is None:
-        timeout = -1
-    else:
+    if isinstance(timeout_sec, float) or isinstance(timeout_sec, int):
```

Comment:
This is not really necessary since the `float` call will already raise in cases where it can't be converted. Without this you can even pass `"1.23"`.